contracts/Governance/ACMCommandsAggregator.sol

```diff
@@ -0,0 +1,152 @@
+
+    /*
+     * @notice 2D array to store grant permissions in batches
+     */
+    Permission[][] public grantPermissions;
+
+    /*
+     * @notice 2D array to store revoke permissions in batches
+     */
+    Permission[][] public revokePermissions;
+
+    /*
+     * @notice Event emitted when grant permissions are added
+     */
+    event GrantPermissionsAdded(uint256 index);
+
+    /*
+     * @notice Event emitted when revoke permissions are added
+     */
+    event RevokePermissionsAdded(uint256 index);
+
+    /*
+     * @notice Event emitted when grant permissions are executed
+     */
+    event GrantPermissionsExecuted(uint256 index);
+
+    /*
+     * @notice Event emitted when revoke permissions are executed
+     */
+    event RevokePermissionsExecuted(uint256 index);
+
+    /*
+     * @notice Error to be thrown when permissions are empty
+     */
+    error EmptyPermissions();
+
+    /*
+     * @notice Constructor to set the access control manager
+     * @param _acm Address of the access control manager
+     */
+    constructor(IAccessControlManagerV8 _acm) {
+        ACM = _acm;
+    }
+
+    /*
+     * @notice Function to add grant permissions
+     * @param _permissions Array of permissions
+     * @custom:event Emits GrantPermissionsAdded event
+     */
+    function addGrantPermissions(Permission[] memory _permissions) external {
+        if (_permissions.length == 0) {
+            revert EmptyPermissions();
+        }
+
+        uint256 index = grantPermissions.length;
+        grantPermissions.push();
+
+        for (uint256 i = 0; i < _permissions.length; i++) {
+            grantPermissions[index].push(
+                Permission(_permissions[i].contractAddress, _permissions[i].functionSig, _permissions[i].account)
+            );
+        }
+
+        emit GrantPermissionsAdded(index);
+    }
+
+    /*
+     * @notice Function to add revoke permissions
+     * @param _permissions Array of permissions
+     * @custom:event Emits RevokePermissionsAdded event
+     */
+    function addRevokePermissions(Permission[] memory _permissions) external {
+        if (_permissions.length == 0) {
+            revert EmptyPermissions();
+        }
+
+        uint256 index = revokePermissions.length;
+        revokePermissions.push();
+
+        for (uint256 i = 0; i < _permissions.length; i++) {
+            revokePermissions[index].push(
+                Permission(_permissions[i].contractAddress, _permissions[i].functionSig, _permissions[i].account)
+            );
+        }
+
+        emit RevokePermissionsAdded(index);
+    }
+
+    /*
+     * @notice Function to execute grant permissions
+     * @param index Index of the permissions array
+     * @custom:event Emits GrantPermissionsExecuted event
+     */
+    function executeGrantPermissions(uint256 index) external {
+        uint256 length = grantPermissions[index].length;
+        for (uint256 i = 0; i < length; i++) {
+            ACM.giveCallPermission(
```

Comment:
Would it be cheaper to save `grantPermissions[index][i]` in a temporary variable?